nixos/modules/services/misc/godns.nix

```diff
@@ -0,0 +1,97 @@
+{ pkgs, lib, config, options, ... }:
+
+with lib;
+
+let
+  cfg = config.services.godns;
+
+  settingsFormat = pkgs.formats.json {};
+  settingsFile = settingsFormat.generate "godns.json" ({
+    ip_url = "https://ip4.seeip.org";
+    ipv6_url = "https://ip6.seeip.org";
+    ip_type = "IPv4";
+    interval = 300;
+    resolver = "8.8.8.8";
+  } // cfg.settings);
+
+in
+{
+  options = {
+    services.godns = {
+      enable = mkEnableOption "GoDNS";
+
+      package = mkOption {
```

Comment:
Unnecessary,  users can just use a overlay since godns isn't used as a dependency in many packages